### DISCUSSION TITLE
IBC Upgrade to latest Version / IBC Hooks

### DISCUSSION THREAD

**Discussion by kevincolahan** [archway-network/discussions#2] [Ideas]
Andromeda is looking to ease our integration as we progress building. For cross-chain integration, our team has a request for Archway to implement IBC Hooks and/or upgrade to the latest version of IBC, v7.3.

Our current implementation uses ibc-hooks but we'd like to support ADR-008 when it becomes available.  More details below on ADR-008.

https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-008-app-caller-cbs/adr-008-app-caller-cbs.md
It's to provide callbacks, the main issue with the current IBC implementation is that sending a wasm message cross-chain with funds attached is unreliable. The ibc-hooks module solves this by attaching the wasm message as a memo for an ics20 transfer and adr-008 can solve this by first sending the ics20 transfer, receiving the "ack-success/failure" response and sending the wasm message separately (for success) or allowing funds recovery (for failure).

Love what you all have built so far and look forward to growing together!

**Comment by zanicar** [comment]
Thank you @kevincolahan  for this submission. This will be discussed in our protocol meeting this week...

**Reply by zanicar** [reply]
If you keep an eye on this repository you'll see that we are upgrading dependencies with every tagged release. However, it may be a while before we get to IBC v7.3 as we have multiple dependencies that must be upgraded along with integration testing and security concerns (including audits) that must be addressed with each step.

**Reply by kevincolahan** [reply]
Thank you for the response and understood. Do you have any timeline on that? Are there other options such as ibc hooks or Axelar gmp?

**Reply by kevincolahan** [reply]
Thank you for the response and understood. Do you have any timeline on that? Are there other options such as ibc hooks or Axelar gmp?

**Reply by zanicar** [reply]
We do have our internal targets regarding upgrade timelines, but due to the complexities of multiple upstream dependencies (including audit timelines, etc.) it would be premature to make any definitive statements on reaching IBC v7.3;

IBC-hooks is currently blocked on one of these upgrade paths (currently in progress), but once we are confident of providing an accurate estimate we will make an announcement...

**Comment by zanicar** [comment]
https://github.com/orgs/archway-network/discussions/27

Archway v5 tagged... Will be deployed to Constantine and mainnet soon!

Includes an upgrade of `ibc-go` to v7...
